[Docs] Describe how KoP reuses Pulsar's Authentication provider (#1195)
streamnative/kop: docs/security.md

   oauth.audience="https://broker.example.com";
```

### Together with Pulsar's authentication

> Since KoP reuses Pulsar's authentication providers for authentication, it can be enabled with other authentication providers at the same time.

Reviewer: ```suggestion
Since KoP reuses Pulsar's authentication providers for authentication, you can KoP's authentication and Pulsar authentication at the same time.
```

> Since KoP reuses Pulsar's authentication providers for authentication, it can be enabled with other authentication providers at the same time.

Reviewer: ```suggestion
Since KoP reuses Pulsar's authentication providers for authentication, you can KoP's authentication providers and Pulsar authentication providers at the same time.
```
```suggestion
Since KoP reuses Pulsar's authentication providers for authentication, it can be enabled with other authentication providers at the same time.
```
